### PR TITLE
Feature/form error handling

### PR DESCRIPTION
**Pull request by Shakikka**
## What does this PR do?

- Adds error handling for zip codes
- Adds error handling for duplicate usernames
- Adds error handling for incorrect usernames
- Adds error messages for these things
- Redirects users to appropriate pages when these errors happen
- Adds testing for these features. plus a few other things

## How should it be tested?

- login with a fake username, and you should be redirected to login screen, and see an error message
- create a new user with a username you know already exists, and you should see an error message after being redirected back to the new user form page.

## What do the future iterations hold?

- Fix errors.

## Linked Issues
- #35
- #36
- #37 
- #

### CODE REVIEW

File: public/sw.js [hunk 1]
```diff
@@ -229,9 +228,37 @@ self.addEventListener('fetch', evt => {
...
+//             cache.put(event.request, response.clone());
+//             return response;
+//           })
+//         );
+//       });
+//     }),
+//   );
+// });
+
+
```
Comment (elizhahn) [inline public/sw.js]: All of this should have been removed with the last PR

File: src/components/Form/Form.js [hunk 2]
```diff
@@ -10,33 +10,49 @@ const Form = ({ createNewUser, loginUser, updateLocation, error }) => {
...
         }
     }
 
+    const zipAuth = zip => {
+        console.log(zip)
+        const canada = /^[ABCEGHJ-NPRSTVXY]\d[ABCEGHJ-NPRSTV-Z][ -]?\d[ABCEGHJ-NPRSTV-Z]\d$/i
+        const us = /^[0-9]{5}(?:-[0-9]{4})?$/
+        console.log((canada.exec(zip) || us.exec(zip)? true : false))
+        return (canada.exec(zip) || us.exec(zip) ? true : false)
+
```
Comment (elizhahn) [inline src/components/Form/Form.js]: should delete these console.logs

File: public/sw.js (hunk 1 above)
Comment (Shakikka) [reply public/sw.js]: I merged them on gh. I'm not sure why they didn't take.

File: src/components/Form/Form.js (hunk 2 above)
Comment (Shakikka) [reply src/components/Form/Form.js]: Will do. Also not sure why this test is failing on travis. It is passing locally.

File: public/sw.js (hunk 1 above)
Comment (elizhahn) [reply public/sw.js]: It seems fixed now thanks

**Pull request merged**